data_test.go

```diff
@@ -74,15 +75,16 @@ func TestParseData(t *testing.T) {
 
 func TestIsPSIPayload(t *testing.T) {
 	pm := newProgramMap()
+	esm := newElementaryStreamMap()
 	var pids []int
 	for i := 0; i <= 255; i++ {
-		if isPSIPayload(uint16(i), pm) {
+		if isPSIPayload(uint16(i), pm, esm) {
 			pids = append(pids, i)
 		}
 	}
 	assert.Equal(t, []int{0, 16, 17, 18, 19, 20, 30, 31}, pids)
 	pm.setUnlocked(uint16(1), uint16(0))
-	assert.True(t, isPSIPayload(uint16(1), pm))
+	assert.True(t, isPSIPayload(uint16(1), pm, esm))
```

Comment:
Could you add a test using `esm.setUnlocked()` like it was done with `pm.setUnlocked(uint16(1), uint16(0))` above?